geometry.py

```
@@ -0,0 +1,160 @@
+import math
+from typing import Tuple, Optional, List
```

Ruff (0.15.6)

[warning] `typing.Tuple` is deprecated, use `tuple` instead

(UP035)